$MirOS: ports/emulators/qemu/patches/patch-gnu-c99-math_h,v 1.2 2005/11/15 12:25:08 tg Exp $
--- gnu-c99-math.h.orig	Thu Dec 21 20:59:00 2006
+++ gnu-c99-math.h	Thu Dec 21 21:41:59 2006
@@ -0,0 +1,38 @@
+/* $MirOS: ports/emulators/qemu/patches/patch-gnu-c99-math_h,v 1.2 2005/11/15 12:25:08 tg Exp $ */
+
+#if (defined(__sun__) || defined(__OpenBSD__)) && defined(__GNUC__)
+
+#include <sys/param.h>
+#if (!defined(MirBSD) || (MirBSD < 0x09A6))
+#define NEED_MBSD_COMPAT
+#else
+/* some NetBSD too, probably */
+#undef NEED_MBSD_COMPAT
+#endif
+
+/*
+ * C99 7.12.3 classification macros
+ * and
+ * C99 7.12.14 comparison macros
+ *
+ * ... do not work on Solaris 10 using GNU CC 3.4.x.
+ * Try to workaround the missing / broken C99 math macros.
+ */
+#include <ieeefp.h>
+
+#ifdef NEED_MBSD_COMPAT
+#define isnormal(x)		(fabs(x) < DBL_EPSILON && !isnan(x) && !isinf(x))
+#endif
+
+#define	isgreater(x, y)		((x) > (y))
+#define	isgreaterequal(x, y)	((x) >= (y))
+#define	isless(x, y)		((x) < (y))
+#define	islessequal(x, y)	((x) <= (y))
+
+#define	isunordered(x,y)	(isnan(x) || isnan(y))
+#ifdef NEED_MBSD_COMPAT
+#define lrintf(x)		((long)rintf(x))
+#define llrintf(x)		((long long)rintf(x))
+#endif
+
+#endif
